CompanyApi/Controllers/CompanyController.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Eventing.Reader;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using Microsoft.AspNetCore.Mvc;
+using CompanyApi.Models;
+namespace CompanyApi.Controllers
+{
+    [ApiController]
+    [Route("api")]
+    public class CompanyController : ControllerBase
+    {
+        private static List<Company> companies = new List<Company>();
+        [HttpPost("companies")]
+        public ActionResult<Company> AddNewCompany([FromBody] Company company)
+        {
+            if (companies.Where(cmp => cmp.Name == company.Name).ToList().Count == 0)
```

Comment:
Very good thought! It is very good to check first! But it is too loooong. Maybe refactor this as an "isXXXX" method is better?